jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/ArquillianAppProvider.java

```diff
@@ -0,0 +1,244 @@
+package org.jboss.arquillian.container.jetty.embedded_12_ee10;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.logging.Logger;
+
+import org.eclipse.jetty.ee10.annotations.AnnotationConfiguration;
+import org.eclipse.jetty.deploy.App;
+import org.eclipse.jetty.deploy.AppProvider;
+import org.eclipse.jetty.deploy.DeploymentManager;
+import org.eclipse.jetty.ee10.plus.webapp.EnvConfiguration;
+import org.eclipse.jetty.ee10.plus.webapp.PlusConfiguration;
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.util.URIUtil;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.resource.Resource;
+import org.eclipse.jetty.ee10.webapp.FragmentConfiguration;
+import org.eclipse.jetty.ee10.webapp.JettyWebXmlConfiguration;
+import org.eclipse.jetty.ee10.webapp.WebAppContext;
+import org.eclipse.jetty.util.resource.ResourceFactory;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+
+public class ArquillianAppProvider extends AbstractLifeCycle implements AppProvider {
+    private static final Logger LOG = Logger.getLogger(ArquillianAppProvider.class.getName());
+
+    /**
+     * The prefix assigned to the temporary file where the archive is exported
+     */
+    private static final String EXPORT_FILE_PREFIX = "export";
+
+    /**
+     * Directory into which we'll extract export the war files
+     */
+    private static final File EXPORT_DIR;
+
+    static {
+        /*
+         * Use of java.io.tmpdir Should be a last-resort fallback for temp directory.
+         * 
+         * Use of java.io.tmpdir on CI systems is dangerous (overwrite possibility is extremely high)
+         * 
+         * Use of java.io.tmpdir on Unix systems is unreliable (due to common /tmp dir cleanup processes)
+         */
+        File systemDefaultTmpDir = new File(AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty("java.io.tmpdir")));
+
+        // If running under maven + surefire, use information provided by surefire.
+        String baseDirVal = AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty("basedir"));
```

Comment:
oh no!

AccessController is deprecated in the Java 17 btw.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/AccessController.html

Reply:
good catch done with cbf57cc01b5edd809a9f0dd15075e7ad9a88c9bc